scripts/bootstrap.sh.in

```diff
@@ -172,37 +172,62 @@ orchestrate_bootstrap() {
     # Grains must be set (in `/etc/salt/grains`) *before* invoking `salt-call`,
     # otherwise grains set during execution won't be taken into account
     # properly.
-    ${SALT_CALL} --local --retcode-passthrough state.sls \
+    ${SALT_CALL} --local --state-output=mixed --retcode-passthrough state.sls \
         metalk8s.node.grains \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION
 
-    # TODO: FIX register IP
+    # TODO: FIX registry IP
     local registry_ip
     registry_ip=$(salt-call --local grains.get metalk8s:control_plane_ip --out txt | awk '/^local\: /{ print $2 }')
-    ${SALT_CALL} --local --retcode-passthrough state.orchestrate \
-        metalk8s.orchestrate.bootstrap_without_master \
+
+    ${SALT_CALL} --local --state-output=mixed  --retcode-passthrough state.sls \
+        metalk8s.roles.bootstrap \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION \
-        pillar="{'registry_ip': $registry_ip}"
-
+        pillar="{'repo': {'local_mode': True}, 'registry_ip': $registry_ip}"
     set_salt_command
     sync_salt
+    # Do CA only first to push informations to the mine
+    ${SALT_CALL} --state-output=mixed --retcode-passthrough state.sls \
+        metalk8s.roles.ca \
+        saltenv=metalk8s-@@VERSION \
+        pillarenv=metalk8s-@@VERSION
+    ${SALT_CALL} --state-output=mixed --retcode-passthrough state.sls \
```

Comment:
Can't this be `highstate`?

Reply:
Can but then we need to give the pillar in the commandline with the roles and the version

Reply:
Yup, why not? :)

Reply:
If we change we will have 2 call with a pillar something like that
```
${SALT_CALL} --state-output=mixed --retcode-passthrough state.highstate pillar="{'metalk8s': {'nodes': {$bootstrap_id: {'roles': ['ca'], 'version': @@VERSION}}}}"
${SALT_CALL} --state-output=mixed --retcode-passthrough state.highstate pillar="{'metalk8s': {'nodes': {$bootstrap_id: {'roles': ['bootstrap'], 'version': @@VERSION}}}, 'registry_ip': $registry_ip}"
```
For me it's ugly to overwrite almost all the pillar in command line.
+ maybe the salt-master address also

Reply:
```
cat > /tmp/pillar.yml << EOF
metalk8s:
  foo:
    bar: ${BAR}
EOF

salt-call ... pillar="$(cat /tmp/pillar.yml)"
```

or something along those lines.

Reply:
(May need to have some way to escape the content of that file for shell injection)

Reply:
I will do it in command line for the moment we will see later to improve it

Reply:
No need for a file, even:
```sh
blah() {
    read -r -d '' PILLAR <<- END
        metalk8s:
          foo:
            bar: '2.0'
    END

    salt-call pillar="$PILLAR"
}
```

Reply:
Yes I will do something like that